### DISCUSSION TITLE
My App does not build on vercel

### DISCUSSION THREAD

**Discussion by Knaackee** [vercel/community#4436] [Help]
### Summary

I can successfully build and run my app locally but the build on vercel fails with the following message:

`Type 'React.ReactNode' is not assignable to type 'import("/vercel/path0/node_modules/@types/react-dom/node_modules/@types/react/ts5.0/index").ReactNode'`

This happens for simple components only returning a div.

It's a typing issue and I tried using “resolutions” but it doesn't work. 

Any ideas?

### Steps to Reproduce

This is hard to reproduce, i can't provide the whole project source.

**Comment by srcnalt** [comment]
Does the issue still happen if you `npm run build` locally?
Also how does it behave if you used `JSX.Element` return type instead?

**Reply by Knaackee** [reply]
Local build does succeed.

**Reply by srcnalt** [reply]
Not entirely sure but could be typescript version related, just a blind guess. I wonder if a fresh example with isolated code deployed, how would it respond? Maybe a code sandbox with exact typescript, react and nextjs versions could help replicate it.

**Comment by Knaackee** [comment]
I just checked out an older version where the deployment was successful, and it failed too?!?

**Comment by Knaackee** [comment]
I solved the problem by updating `@types/react`
